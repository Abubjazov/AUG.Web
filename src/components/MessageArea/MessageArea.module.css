.root {
  position: absolute;
  top: 10px;
  left: calc(50% - 150px);
  z-index: 100;
  display: flex;
  flex-direction: column;
  gap: 5px;
  align-items: flex-start;
  justify-content: flex-start;
  width: 300px;
  padding: 0;
  background: transparent;
  animation: var(--animation-visible);
}
