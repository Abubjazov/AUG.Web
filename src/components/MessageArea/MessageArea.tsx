import { FC } from 'react'

import { useAppSelector } from 'store/hooks'
import Snackbar from 'uikit/Snackbar'

import styles from './MessageArea.module.css'

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface MessageAreaProps {}

const MessageArea: FC<MessageAreaProps> = () => {
  const { messages } = useAppSelector(state => state.layout)

  return (
    <div className={styles['root']}>
      {messages.map(message => (
        <Snackbar key={message.messageId} message={message} />
      ))}
    </div>
  )
}

export default MessageArea
