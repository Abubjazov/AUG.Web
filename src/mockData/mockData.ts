import { IDapplet, ITag } from 'store/slices/dappletsSlice'
import { IList, IUserDapplet } from 'store/slices/userDataSlice'

export const mockUserTags: ITag[] = [
  { tagId: 'darP5Jyz8yirTMsfY9RP', tagName: 'Twitter' },
  { tagId: 'JW3UFtZ5HgATcwldsJ1T', tagName: 'Facebook' },
  { tagId: '6UKxNrMzte6RCWIlssvM', tagName: 'Top 10' },
  { tagId: 'ZbEuko9nOlQYKG6I1Efx', tagName: 'Test' },
  { tagId: '521keF1Adymn8Oo896vv', tagName: 'ToDo' },
]

export const mockUserLists: IList[] = [
  { listId: 'XfLdXIykGHsgIjdJkTaE', listName: 'My special list' },
  { listId: 'c6rClgOltChco2XnyVDp', listName: 'Financial list' },
]

export const mockUserDapplets: IUserDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    userTags: ['JW3UFtZ5HgATcwldsJ1T', '521keF1Adymn8Oo896vv'],
    dappletState: true,
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    userTags: [],
    dappletState: true,
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    userTags: 'N/A',
    dappletState: true,
  },
]

export const mockCommunityTags: ITag[] = [
  { tagId: 'XoEVTHXhfMAjIWsi4wxY', tagName: 'Top 100' },
  { tagId: '3LYUFQTbdNULwBgCkfXx', tagName: 'Finances' },
  { tagId: 'XZ2zPZqa4fPxM9mLA9xR', tagName: 'Media' },
  { tagId: 'C69B5fQ7ccW6Q1SC0yow', tagName: 'Favourites' },
]

export const mockDapplets: IDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    appOwner: 'Satoshi Nakamoto',
    circulatingSupply: '19534212',
    communityTags: ['3LYUFQTbdNULwBgCkfXx', 'C69B5fQ7ccW6Q1SC0yow'],
    date: '1230757200',
    fullDesc:
      'Bitcoin is a peer-to-peer online currency, meaning that all transactions happen directly between equal, independent network participants, without the need for any intermediary to permit or facilitate them. Bitcoin was created, according to Nakamoto’s own words, to allow “online payments to be sent directly from one party to another without going through a financial institution.”',
    fullyDilutedMarketCap: '738866583732',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-1.01',
    maxSupply: '21000000',
    name: 'Bitcoin Contracts',
    shortDesc:
      'Bitcoin’s original inventor is known under a pseudonym, Satoshi Nakamoto. ',
    shortName: 'BTC',
    totalSupply: '19534106',
    volume: '10.9',
    volumePerMarketCap: '1.58',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHGCOPY',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    appOwner: 'XRP Ledger',
    circulatingSupply: '53615837759',
    communityTags: 'N/A',
    date: '1620939600',
    fullDesc:
      'Launched in 2021, the XRP Ledger (XRPL) is an open-source, permissionless and decentralized technology. Benefits of the XRP Ledger include its low-cost ($0.0002 to transact), speed (settling transactions in 3-5 seconds), scalability (1,500 transactions per second) and inherently green attributes (carbon-neutral and energy-efficient). The XRP Ledger also features the first decentralized exchange (DEX) and custom tokenization capabilities built into the protocol. Since 2012, the XRP Ledger has been operating reliably, having closed 70 million ledgers.',
    fullyDilutedMarketCap: '65183420834',
    logo: '/public/images/notAvailable.svg',
    marketCap: '5.66',
    maxSupply: '100000000000',
    name: 'XRP Ledger',
    shortDesc:
      'Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.',
    shortName: 'XRP',
    totalSupply: '99988316618',
    volume: '132.26',
    volumePerMarketCap: '5.37',
  },
  {
    dappletId: 'QbWG3sKvfgFcP5RtskMp',
    appOwner: 'Realcoin',
    circulatingSupply: '85367759449',
    communityTags: [],
    date: '1388523600',
    fullDesc:
      'Launched in 2014, Tether is a blockchain-enabled platform designed to facilitate the use of fiat currencies in a digital manner. Tether works to disrupt the conventional financial system via a more modern approach to money. Tether has made headway by giving customers the ability to transact with traditional currencies across the blockchain, without the inherent volatility and complexity typically associated with a digital currency. As the first blockchain-enabled platform to facilitate the digital use of traditional currencies (a familiar, stable accounting unit), Tether has democratised cross-border transactions across the blockchain.',
    fullyDilutedMarketCap: '88708606718',
    logo: '/dappletsLogos/tether.png',
    marketCap: '0.01',
    maxSupply: 'N/A',
    name: 'Tether USDt',
    shortDesc:
      'Tether works to disrupt the conventional financial system via a more modern approach to money.',
    shortName: 'USDT',
    totalSupply: '88623656724',
    volume: '41.44',
    volumePerMarketCap: '32.16',
  },
]
