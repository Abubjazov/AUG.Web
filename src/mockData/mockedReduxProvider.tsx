import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import {
  mockCommunityTags,
  mockDapplets,
  mockMyDapplets,
  mockMyTags,
} from './mockData'
import dappletsReducer from '../store/slices/dappletsSlice'
import layoutReducer from '../store/slices/layoutSlice'
import myDappletsReducer from '../store/slices/myDappletsSlice'

export const defaultMockState = {
  layout: {
    menuOpened: false,
    dappletSettingsOpened: false,
  },

  myDapplets: {
    myDapplets: [...mockMyDapplets],
    myTags: [...mockMyTags],
  },

  dapplets: {
    dapplets: [...mockDapplets],
    tags: [...mockCommunityTags],
  },
}

export const mockedReduxProvider = ({
  children,
}: PropsWithChildren<object>) => {
  const store = configureStore({
    reducer: {
      layout: layoutReducer,
      dapplets: dappletsReducer,
      myDapplets: myDappletsReducer,
    },
    preloadedState: {
      ...defaultMockState,
    },
  })

  return <Provider store={store}>{children}</Provider>
}
