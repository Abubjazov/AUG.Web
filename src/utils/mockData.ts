export const mockMyTags = [
  { tagId: 1, tagName: 'Twitter' },
  { tagId: 2, tagName: 'Facebook' },
  { tagId: 3, tagName: 'Top 10' },
  { tagId: 4, tagName: 'Test' },
  { tagId: 5, tagName: 'ToDo' },
]

export const mockCommunityTags = [
  { tagId: 1, tagName: 'Top 100' },
  { tagId: 2, tagName: 'Finances' },
  { tagId: 3, tagName: 'Media' },
  { tagId: 4, tagName: 'Favourites' },
]

export const mockDapplets = [
  {
    dappletId: 1,
    logo: '/public/mockData/logos/dyson.svg',
    name: 'Ethereum Contracts Example',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'debra.holt',
    communityTags: [
      { tagId: 1, tagName: 'Top 100' },
      { tagId: 2, tagName: 'Social' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 2,
    logo: '/public/mockData/logos/bmw.svg',
    name: 'Habitant sollicitudin velit',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'michael.mitc',
    communityTags: [
      { tagId: 2, tagName: 'Social' },
      { tagId: 4, tagName: 'Favourites' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 3,
    logo: '/public/mockData/logos/dyson.svg',
    name: 'Ethereum Contracts Example',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'sara.cruz',
    communityTags: [
      { tagId: 1, tagName: 'Top 100' },
      { tagId: 2, tagName: 'Social' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 4,
    logo: '/public/mockData/logos/bmw.svg',
    name: 'Habitant sollicitudin velit',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'jessica.hanson',
    communityTags: [
      { tagId: 2, tagName: 'Social' },
      { tagId: 4, tagName: 'Favourites' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 5,
    logo: '/public/mockData/logos/dyson.svg',
    name: 'Ethereum Contracts Example',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'debra.holt',
    communityTags: [
      { tagId: 1, tagName: 'Top 100' },
      { tagId: 2, tagName: 'Social' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 6,
    logo: '/public/mockData/logos/bmw.svg',
    name: 'Habitant sollicitudin velit',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'michael.mitc',
    communityTags: [
      { tagId: 2, tagName: 'Social' },
      { tagId: 4, tagName: 'Favourites' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 7,
    logo: '/public/mockData/logos/dyson.svg',
    name: 'Ethereum Contracts Example',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'sara.cruz',
    communityTags: [
      { tagId: 1, tagName: 'Top 100' },
      { tagId: 2, tagName: 'Social' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 8,
    logo: '/public/mockData/logos/bmw.svg',
    name: 'Habitant sollicitudin velit',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'jessica.hanson',
    communityTags: [
      { tagId: 2, tagName: 'Social' },
      { tagId: 4, tagName: 'Favourites' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 9,
    logo: '/public/mockData/logos/dyson.svg',
    name: 'Ethereum Contracts Example',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'debra.holt',
    communityTags: [
      { tagId: 1, tagName: 'Top 100' },
      { tagId: 2, tagName: 'Social' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 10,
    logo: '/public/mockData/logos/bmw.svg',
    name: 'Habitant sollicitudin velit',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'michael.mitc',
    communityTags: [
      { tagId: 2, tagName: 'Social' },
      { tagId: 4, tagName: 'Favourites' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 11,
    logo: '/public/mockData/logos/dyson.svg',
    name: 'Ethereum Contracts Example',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'sara.cruz',
    communityTags: [
      { tagId: 1, tagName: 'Top 100' },
      { tagId: 2, tagName: 'Social' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 12,
    logo: '/public/mockData/logos/bmw.svg',
    name: 'Habitant sollicitudin velit',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'jessica.hanson',
    communityTags: [
      { tagId: 2, tagName: 'Social' },
      { tagId: 4, tagName: 'Favourites' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
  {
    dappletId: 13,
    logo: '/public/mockData/logos/dyson.svg',
    name: 'Ethereum Contracts Example',
    date: 1691942217,
    shortDesc: 'Feature adds tweets to Ethereum contract',
    fullDesc:
      'Semper neque leo scelerisque gravida pharetra, elit viverra varius. Leo et pretium massa tristique mauris habitasse in in. Et cras sociis tellus viverra at dictumst quisque praesent arcu. Eu faucibus id at odio praesent. Ut nibh porta ipsum in eget id netus pharetra in. Ultrices scelerisque augue dui eget lacus, aenean cursus in. Consequat rhoncus egestas ultricies imperdiet diam. Imperdiet mollis egestas est faucibus sit tristique eu.',
    appOwner: 'sara.cruz',
    communityTags: [
      { tagId: 1, tagName: 'Top 100' },
      { tagId: 2, tagName: 'Social' },
    ],
    semperNeque: 'Amet, porttitor gravida id',
    aliquam: 'Viverra dictum arcu.',
    urna: 'Ultrices pellentesque.',
    leoIpsum: 'Vulputate dolor nam in.',
    inEuismod: 'Felis pretium neque.',
    namDiam: 'Elit accumsan leo non.',
    elitSagittis: 'Ut molestie eros.',
    justoAmet: 'Nullam nec ultrices.',
  },
]

export const mockMyDapplets = [
  {
    dappletId: 1,
    userTags: [
      { tagId: 1, tagName: 'Twitter' },
      { tagId: 5, tagName: 'ToDo' },
    ],
    dappletState: true,
  },

  {
    dappletId: 4,
    userTags: [
      { tagId: 1, tagName: 'Twitter' },
      { tagId: 2, tagName: 'Facebook' },
    ],
    dappletState: false,
  },

  {
    dappletId: 6,
    userTags: [
      { tagId: 1, tagName: 'Twitter' },
      { tagId: 4, tagName: 'Test' },
      { tagId: 5, tagName: 'ToDo' },
    ],
    dappletState: true,
  },
]
