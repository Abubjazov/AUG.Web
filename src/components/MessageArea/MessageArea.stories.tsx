import { configureStore } from '@reduxjs/toolkit'
import type { Meta, StoryObj } from '@storybook/react'
import MockedProvider, { defaultMState } from 'mockData/mockedReduxProvider'
import { AppStore } from 'store/index'
import layoutReducer, { EMessageType } from 'store/slices/layoutSlice'

import MessageArea, { MessageAreaProps } from './MessageArea'

import '/src/index.css'

const meta: Meta<MessageAreaProps> = {
  component: MessageArea,
  title: 'COMPONENTS/Message Area',
  parameters: {
    docs: {
      description: {
        component:
          'Interface component - MessageArea. Component for visualizing service messages and errors. Contains such components as: Snackbar',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    showMode: {
      description:
        'Activate show mode for StoryBook (deactivates self-destruction of messages from the message queue)',
    },
  },
}

export default meta

type Story = StoryObj<MessageAreaProps>

const mockedStore: AppStore = configureStore({
  reducer: {
    layout: layoutReducer,
  },
  preloadedState: {
    layout: {
      ...defaultMState.layout,
      messages: [
        {
          messageId: '13',
          messageText: 'ERROR Message text',
          messageType: EMessageType.ERROR,
        },

        {
          messageId: '31',
          messageText: 'INFO Message text',
          messageType: EMessageType.INFO,
        },
      ],
    },
  },
})

export const Default: Story = {
  args: {
    showMode: true,
  },
  decorators: [
    Story => (
      <MockedProvider mockedStore={mockedStore}>
        <div
          style={{
            height: '100px',
          }}
        >
          <Story />
        </div>
      </MockedProvider>
    ),
  ],
}
