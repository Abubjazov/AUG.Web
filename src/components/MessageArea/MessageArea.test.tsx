import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { render } from '@testing-library/react'
import { mockedReduxProvider as MProvider } from 'mockData/mockedReduxProvider'
import { Provider } from 'react-redux'
import layoutReducer, { EMessageType } from 'store/slices/layoutSlice'

import MessageArea from './MessageArea'

describe('MessageArea', () => {
  test('should render MessageArea default', () => {
    const { asFragment } = render(
      <MProvider>
        <MessageArea />
      </MProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render MessageArea with messages', () => {
    const mokedStore = configureStore({
      reducer: {
        layout: layoutReducer,
      },
      preloadedState: {
        layout: {
          menuOpened: true,
          menuButtonsState: 0,
          dappletSettingsOpened: false,
          modalState: false,
          modalInner: undefined,
          messages: [
            {
              messageId: '13',
              messageText: 'ERROR Message text',
              messageType: EMessageType.ERROR,
            },

            {
              messageId: '31',
              messageText: 'INFO Message text',
              messageType: EMessageType.INFO,
            },
          ],
        },
      },
    })

    const NewProvider = ({ children }: PropsWithChildren<object>) => {
      return <Provider store={mokedStore}>{children}</Provider>
    }

    const { asFragment } = render(
      <NewProvider>
        <MessageArea />
      </NewProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
