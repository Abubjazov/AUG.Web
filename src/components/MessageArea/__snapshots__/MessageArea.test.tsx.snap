// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MessageArea > should render MessageArea default 1`] = `
<DocumentFragment>
  <div
    class="_root_710b57"
  />
</DocumentFragment>
`;

exports[`MessageArea > should render MessageArea with messages 1`] = `
<DocumentFragment>
  <div
    class="_root_710b57"
  >
    <div
      class="_root_4e3dca _error_4e3dca "
      data-testid="snackbar"
    >
      <span
        class="_title_4e3dca"
      >
        ERROR Message text
      </span>
    </div>
    <div
      class="_root_4e3dca _info_4e3dca "
      data-testid="snackbar"
    >
      <span
        class="_title_4e3dca"
      >
        INFO Message text
      </span>
    </div>
  </div>
</DocumentFragment>
`;
